source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -504,11 +503,7 @@ Http::FilterHeadersStatus JsonTranscoderFilter::encodeHeaders(Http::ResponseHead
   }
 
   headers.setReferenceContentType(Http::Headers::get().ContentTypeValues.Json);
-  if (!method_->descriptor_->server_streaming()) {
-    return Http::FilterHeadersStatus::StopIteration;
-  }
-
-  return Http::FilterHeadersStatus::Continue;
+  return Http::FilterHeadersStatus::StopIteration;
```

Comment:
I'm not sure that this is correct way of solving problem:

When filter returns with `Http::FilterHeadersStatus::Continue` it automatically adds(preserves?) `Content-Type: application/grpc` header.

I wasn't able to find correct(?) way of doing this, could you please confirm / advice a better way?

Reply:
I think returning "Stop" is be fine. It needs to wait for encodeData() call to set the header content-type.   If continue, it will be too late for encodeData() to modify the headers

Reply:
Thanks for feedback!
Updated accordingly.